tests/test_project.py

```diff
@@ -861,6 +970,51 @@ def get_sp(i):
                 assert str(job) == k
         assert group_count == len(list(self.project.find_jobs()))
 
+        # using sp.key and doc.key
+        for k, g in self.project.groupby('sp.a'):
+            assert len(list(g)) == 1
+            for job in list(g):
+                assert job.sp['a'] == k
+
+        assert len(list(self.project.groupby('d'))) == 0
+        for k, g in self.project.groupby('sp.d', default=-1):
```

Comment:
It is not clear to me how these two lines relate to each other. Is the for-loop even entered at all if the length of that grouping is zero?

Reply:
As `d` is not present in jobs, `self.project.groupby('sp.d', default=-1)` will return one group having key as `default` (which is -1).

Reply:
Got it, thx.